pkg/controllers/status/cluster_status_controller.go

```diff
@@ -194,7 +194,9 @@ func (c *ClusterStatusController) syncClusterStatus(cluster *clusterv1alpha1.Clu
 		// 	We should move this logic to both `execution-controller` and `work-status-controller`.
 		//  After that the 'initializeGenericInformerManagerForCluster' function as well as 'c.GenericInformerManager'
 		//  can be safely removed from current controller.
-		c.initializeGenericInformerManagerForCluster(clusterClient)
+		if err := c.initializeGenericInformerManagerForCluster(clusterClient); err != nil {
```

Comment:
initializeGenericInformerManagerForCluster could return an error, should requeue it.